flow/project.py

```diff
@@ -1594,18 +1594,56 @@ def _print_progress(x):
 
         with self._potentially_buffered():
             try:
-                with contextlib.closing(ThreadPool()) as pool:
-                    _map = map if no_parallelize else pool.imap
-                    # First attempt at parallelized status determination.
-                    # This may fail on systems that don't allow threads.
+                status_parallelization = self.config.get('flow').get('status_parallelization')
+                if status_parallelization == 'thread':
+                    with contextlib.closing(ThreadPool()) as pool:
+                        # First attempt at parallelized status determination.
+                        # This may fail on systems that don't allow threads.
+                        results = pool.imap(_get_job_status, jobs)
+                        return list(tqdm(
+                            iterable=results,
+                            desc="Collecting job status info", total=len(jobs), file=err))
+                elif status_parallelization == 'process':
+                    with contextlib.closing(Pool()) as pool:
+                        try:
+                            import pickle
+                            results = self._fetch_status_in_parallel(
+                                pool, pickle, jobs, ignore_errors, cached_status)
+                        except Exception as error:
+                            if not isinstance(error, (pickle.PickleError, self._PickleError)) and\
+                                    'pickle' not in str(error).lower():
+                                raise    # most likely not a pickle related error...
+
+                            try:
+                                import cloudpickle
+                            except ImportError:  # The cloudpickle package is not available.
+                                logger.error("Unable to parallelize execution due to a "
+                                             "pickling error. "
+                                             "\n\n - Try to install the 'cloudpickle' package, "
+                                             "e.g., with 'pip install cloudpickle'!\n")
+                                raise error
+                            else:
+                                try:
+                                    results = self._fetch_status_in_parallel(
+                                        pool, cloudpickle, jobs, ignore_errors, cached_status)
+                                except self._PickleError as error:
+                                    raise RuntimeError(
+                                        "Unable to parallelize execution due to a pickling "
+                                        "error: {}.".format(error))
+                        return list(tqdm(
+                            iterable=results,
+                            desc="Collecting job status info", total=len(jobs), file=err))
+                else:
```

Comment:
This means that faulty configuration values will use no parallelism if a faulty value is set. I think we should error in this case.

Reply:
Here I've used else so every other case comes here. I don't think we should raise error unless we just want this to be a two-way procedure, i.e. `Process` and `Thread`.

Reply:
If we have a third option `none` then we can have an `elif` and an error raising `else`. This is better, I think, than just accepting any input besides `thread` and `process` as meaning no parallelization. We should not do something the user might find unexpected and regresses performance.

Reply:
Yes, I think that would be a nice alternative. Do you want me to write tests for it too?

Reply:
That would be good. A test for error raising would also be nice.

Reply:
Pushed the changes which resolve this. You can have a look.